lib/Controller/MainController.php

```diff
@@ -218,6 +218,34 @@ public function category($category) {
 		}
 	}
 
+	/**
+	 * @NoAdminRequired
+	 * @NoCSRFRequired
+	 */
+	public function categoryUpdate($category) {
+		$this->dbCacheService->triggerCheck();
+
+		$json = $this->restParser->getParameters();
+		if (!$json || !isset($json['name']) || !$json['name']) {
+			throw new Exception('New category name not found');
+		}
+
+		$category = urldecode($category);
+		try {
+			$recipes = $this->service->getRecipesByCategory($category);
+			foreach ($recipes as $i => $recipe) {
+				$r = $this->service->getRecipeById($recipe['recipe_id']);
+				$r['recipeCategory'] = $json['name'];
+				$this->service->addRecipe($r);
+			}
+
+			return new DataResponse($json['name'], Http::STATUS_OK, ['Content-Type' => 'application/json']);
+		} catch (\Exception $e) {
+			error_log($e);
```

Comment:
Logging to the system log might be better off with the NC logfile, don't you think?

Reply:
I deleted it

Reply:
removed